### PR TITLE
python38Packages.pytwitchapi: init at 2.3.2

### PR DESCRIPTION
**Pull request by WolfangAukang**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133871` at 25ee8bf5 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.pyTwitchAPI</li>
    <li>python39Packages.pyTwitchAPI</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `pytestCheckHook`.
Near pkgs/development/python-modules/pyTwitchAPI/default.nix:4:3:
```
  |
4 | , pytestCheckHook
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/pyTwitchAPI/default.nix:36:0:
```
   |
36 |     description = "A Python 3.7 implementation of the Twitch API and its Webhook";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyTwitchAPI/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+, fetchFromGitHub
+, pythonOlder
+, aiohttp
+, python-dateutil
+, requests
+, websockets
+}:
+
+buildPythonPackage rec {
+  pname = "pyTwitchAPI";
```
Comment (fabaff) [inline pkgs/development/python-modules/pyTwitchAPI/default.nix:12]: Should follow [PEP-503](https://nixos.org/manual/nixpkgs/stable/#contributing).

```suggestion
  pname = "pytwitchapi";
```

File: pkgs/development/python-modules/pyTwitchAPI/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  propagatedBuildInputs = [
+    aiohttp
+    python-dateutil
+    requests
+    websockets
+  ];
+
+  doChecks = false;
+
+  pythonImportsCheck = [ "aiohttp" "dateutil" "requests" "twitchAPI" "websockets" ];
```
Comment (fabaff) [inline pkgs/development/python-modules/pyTwitchAPI/default.nix:32]: No need to check the import of the used modules.

```suggestion
  pythonImportsCheck = [ "twitchAPI" ];
```

File: pkgs/development/python-modules/pyTwitchAPI/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  propagatedBuildInputs = [
+    aiohttp
+    python-dateutil
+    requests
+    websockets
+  ];
+
+  doChecks = false;
```
Comment (fabaff) [inline pkgs/development/python-modules/pyTwitchAPI/default.nix:30]: Please add a comment why we don't run the tests.

File: pkgs/development/python-modules/pyTwitchAPI/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  pythonImportsCheck = [ "aiohttp" "dateutil" "requests" "twitchAPI" "websockets" ];
+
+  meta = with lib; {
+    description = "A Python 3.7 implementation of the Twitch API and its Webhook";
```
Comment (fabaff) [inline pkgs/development/python-modules/pyTwitchAPI/default.nix:35]: According to the `setup.py` file.

```suggestion
    description = "Python implementation of the Twitch Helix API, its webhook and PubSub";
```

File: pkgs/development/python-modules/pyTwitchAPI/default.nix (hunk 1 above)
Comment (WolfangAukang) [reply pkgs/development/python-modules/pyTwitchAPI/default.nix]: Changes implemented

File: pkgs/development/python-modules/pyTwitchAPI/default.nix (hunk 2 above)
Comment (WolfangAukang) [reply pkgs/development/python-modules/pyTwitchAPI/default.nix]: Changes implemented

File: pkgs/development/python-modules/pyTwitchAPI/default.nix (hunk 3 above)
Comment (WolfangAukang) [reply pkgs/development/python-modules/pyTwitchAPI/default.nix]: Changes implemented

File: pkgs/development/python-modules/pyTwitchAPI/default.nix (hunk 4 above)
Comment (WolfangAukang) [reply pkgs/development/python-modules/pyTwitchAPI/default.nix]: Changes implemented

**Pull request merged**